prow/genjobs/pkg/util/os.go

```diff
@@ -25,3 +26,8 @@ func RenameFile(pat string, src string, repl string) string {
 	s := regexp.MustCompile(pat).ReplaceAllString(src, repl)
 	return regexp.MustCompile(`^[^\w\d]`).ReplaceAllString(s, "")
 }
+
+// HasExtension checks if a file's extension matches a pattern.
+func HasExtension(path string, pat string) bool {
+	return regexp.MustCompile(pat).MatchString(filepath.Ext(path))
```

Comment:
This will cause a runtime panic if `pat` is not a valid regexp. Please either compile the regexp at init time or return an error.

Reply:
I am using `regexp.MustCompile(..)` in *several* other places, so I will need to refactor the code heavily. I will tackle this issue with a robust solution in a follow-up PR #2110; this diff is already quite large?